chart/compass/charts/connector/templates/service-role-binding.yaml

```diff
@@ -11,27 +11,22 @@ metadata:
     app.kubernetes.io/managed-by: {{ .Release.Service }}
     app.kubernetes.io/instance: {{ .Release.Name }}
 spec:
+  selector:
+    matchLabels:
+      app: {{ .Chart.Name }}
+      release: {{ .Release.Name }}
+  action: ALLOW
   rules:
-    - services: ["{{ template "fullname" . }}-internal.{{ .Release.Namespace }}.svc.cluster.local"]
-      paths: ["*"]
-      methods: ["*"]
----
-apiVersion: "rbac.istio.io/v1alpha1"
-kind: ServiceRoleBinding
-metadata:
-  name: {{ template "fullname" . }}-internal
-  namespace: {{ .Release.Namespace }}
-  labels:
-    release: {{ .Release.Name }}
-    helm.sh/chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
-    app.kubernetes.io/name: {{ template "name" . }}
-    app.kubernetes.io/managed-by: {{ .Release.Service }}
-    app.kubernetes.io/instance: {{ .Release.Name }}
-spec:
-  subjects:
-{{ tpl (toYaml .Values.istio.rbac.connectorInternalApi.subjects | indent 4) . }}
-  roleRef:
-    kind: ServiceRole
-    name: {{ template "fullname" . }}-internal
----
+    - from:
+      - source:
+          principals:
+          - {{ tpl .Values.istio.rbac.connectorInternalApi.compassDirector . }}
+          - {{ tpl .Values.istio.rbac.connectorInternalApi.compassConnectorTests . }}
+    - to:
```

Comment:
It looks to me (but i might be wrong) that this is more restrictive that what was previously defined. Right now we will restrict access to all connector apis (external and internal) to be accessible only by the director and compass tests service acccounts.
So im wondering with this change will apps that want to pair with compass and obtain CSR configuration details from connector be able to request it ? (we have two cases - the app can be running in the same cluster in different namespace (or in the same namespace as well i guess for test purposes?) or outside of the cluster altogether

Reply:
Previously, the `compass-connector` application was secured with the [ClusterRbacConfig](https://github.com/kyma-project/kyma/blob/master/resources/core/charts/istio-rbac/templates/rbac-config.yaml) for which the value is defined [here](https://github.com/kyma-project/kyma/blob/master/resources/core/charts/istio-rbac/values.yaml#L8). As you can see it secures only one service - `compass-connector-internal`. For access to that service, the ServiceRole was created which gives an access to all methods and all paths. This ServiceRole was bounded to `cluster.local/ns/{{ .Release.Namespace }}/sa/compass-director ` and `cluster.local/ns/{{ .Release.Namespace }}/sa/compass-connector-tests` serviceaccounts with the ServiceRoleBinding.

Again, please notice that the only service which was secured with RBAC was `compass-connector-internal`. You can find it [here](https://github.com/kyma-incubator/compass/blob/78c43b3f308b41b3e9af08c4f2f3a19b75494660/chart/compass/charts/connector/templates/service.yaml#L27). The ports used by this service are `{{ .Values.global.connector.graphql.internal.port }}`  and `{{ .Values.global.istio.proxy.port }}`, so the access to these ports should be possible only for serviceaccounts mentioned above. And all the remaining ports should be open to anyone.

Now what this new AuthorizationPolicy does is:
1) allow the access to the whole application (any ports) for two defined serviceaccounts with this code:
```
    - from:
      - source:
          principals:
          - {{ tpl .Values.istio.rbac.connectorInternalApi.compassDirector }}
          - {{ tpl .Values.istio.rbac.connectorInternalApi.compassConnectorTests }}
```
2) allow the access to the particular ports for anyone:
```
    - to:
      - operation:
          ports:
          - "{{ .Values.global.connector.graphql.internal.port }}"
          - "{{ .Values.global.istio.proxy.port }}"
          - "{{ .Values.global.connector.validator.port }}"
          - "{{ .Values.global.connector.graphql.external.port }}"
```

Actually, there is a mistake in `2)`, as the only ports that should be present, should be all ports that are not exposed by `compass-connector-internal` service. So the proper list is:
```
          - "{{ .Values.global.connector.validator.port }}"
          - "{{ .Values.global.connector.graphql.external.port }}"
```

Please notice two things:
- In contrary to ServiceRole the new AuthorizationPolicy applies to the workload (Pod), not to the particular service.
- `to` and `from` defines separate rules in this particular case, as both are the separate objects on the list:
```
- from:
   ...
- to:
   ...
```
the situation you described would happen if these were the properties of the same object:
```
- from:
   ...
   to:
   ...
```

I hope that this explanation is good enough. In case of any question, I will be happy to answer them.

Reply:
thanks, great explanation! So its actually two separate rules - one has just a source, saying this source can access to the whole app, the second rule has only destination/operation, saying that any source can access this destination/operation (public apis).

And if i understand correctly, the correction that is needed to remove the internal ports from the second rule's destination so that they are not accessible by everyone.

Reply:
Exactly!